File: scripts/find_missing_images.py
from pathlib import Path
import re
root = Path('src/data')
text = (root/'products.js').read_text(encoding='utf-8')
product_names = re.findall(r"name:\s*['\"]([^'\"]+)['\"]", text)
local_section = re.search(r"const localImages = \{([\s\S]*?)\};", text)
local_keys = set(re.findall(r"['\"]([^'\"]+)['\"]\s*:", local_section.group(1))) if local_section else set()
missing = [n for n in product_names if n not in local_keys]
files=[]
for dir in ['src/images/gloves','src/images/knives','src/images/weapons']:
    p=Path(dir)
    if p.exists():
        for f in p.iterdir():
            if f.is_file():
                files.append((dir,f.name))

def norm(name):
    return re.sub(r'[^a-z0-9]+',' ', name.lower().replace('|','')).strip()
for name in missing:
    n=norm(name)
    candidates=[(d,f) for d,f in files if n in norm(f) or norm(f) in n or n.split(' ')[-1] in norm(f)]
    if candidates:
        print('MISSING:',name)
        for d,f in candidates[:5]:
            print('  ',d,f)

File: scripts/find_missing_images.mjs
import fs from 'fs';
import path from 'path';
const root = path.resolve('src/data');
const data = fs.readFileSync(path.join(root, 'products.js'), 'utf8');
const productNames = [...data.matchAll(/name:\s*['"]([^'"]+)['"]/g)].map((m) => m[1]);
const localSection = data.match(/const localImages = \{([\s\S]*?)\};/);
const localKeys = new Set();
if (localSection) {
  for (const m of localSection[1].matchAll(/['"]([^'"]+)['"]\s*:/g)) {
    localKeys.add(m[1]);
  }
}
const missing = productNames.filter((n) => !localKeys.has(n));
const files = [];
for (const dir of ['src/images/gloves', 'src/images/knives', 'src/images/weapons']) {
  const p = path.resolve(dir);
  if (fs.existsSync(p)) {
    for (const f of fs.readdirSync(p)) {
      const full = path.join(p, f);
      if (fs.statSync(full).isFile()) {
        files.push({dir, name: f});
      }
    }
  }
}
const norm = (name) => name.toLowerCase().replace(/\|/g, '').replace(/[^a-z0-9]+/g, ' ').trim();
for (const name of missing) {
  const n = norm(name);
  const candidates = files.filter((f) => {
    const fn = norm(f.name);
    return fn.includes(n) || n.includes(fn) || fn.includes(n.split(' ').slice(-1)[0]);
  });
  if (candidates.length) {
    console.log('MISSING:', name);
    for (const c of candidates.slice(0, 5)) {
      console.log('  ', c.dir, c.name);
    }
  }
}
